packages/cxl-ui/src/components/cxl-icon-nav-item.js

```diff
@@ -0,0 +1,32 @@
+// Import the LitElement base class and html helper function
+import { LitElement, html } from 'lit-element';
+import '@vaadin/vaadin-item';
+import cxlThemeStyles from '../styles/cxl-theme-css.js';
+import cxlIconNavItemStyles from '../styles/cxl-icon-nav-item-css.js';
+
+class CXLIconNavItemElement extends LitElement {
+  static get properties() {
+    return {
+      href: {
+        type: String
+      },
+      icon: {
+        type: String
+      }
+    };
+  }
+
+  static get styles() {
+    return [cxlThemeStyles, cxlIconNavItemStyles];
+  }
+
+  render() {
+    return html`
+      <a href="${this.icon}">
+        <iron-icon class="icon size-l" icon="${this.icon}"></iron-icon>
```

Comment:
I'd actually prefer icons to be fetched from element content. `<cxl-icon-nav-item href="#">cxl:training</cxl-icon-nav-item>`. Are there any drawbacks?

`${this.innerHTML}` seems to work without side-effects.

Reply:
@chanar I force-pushed a set of updates.

Reply:
Hmm, lets then use element content for initials (text)/image (imgsrc)/icon (icon name). and just use content-type="imagesrc/text/icon"

Reply:
This proposal is hard to decipher. Can you provide specific markup examples for every member of our scenario set?

Reply:
```
  render() {
    return html`
      <a href=${ifDefined(this.href)}>
        ${this.getNavItemContent}
      </a>
    `;
  }

  getNavItemContent() {

    switch(this.contentType) {
      case 'image':
        return html`<img src="${this.innerHTML}" />`
        break;
      case 'text':
        return html`<span>${$this.innerHTML}</span>`
        break;
      default:
        return html`<iron-icon class="icon size-l" icon="${this.innerHTML}"></iron-icon>`
    }

  }
```

Reply:
```
<vaadin-tab>
            <cxl-icon-nav-item href="#" content-type="icon">cxl:nav-discover</cxl-icon-nav-item>
          </vaadin-tab>
```

Reply:
Nice in theory, but the problem with this is requiring PHP implementation of content-type.

Could you make this work via `<vaadin-tab class>` container attribute? It is difficult to adjust anything deeper than that quickly.

Reply:
So instead content-type="icon" I would make 

`<vaadin-tab class="icon/image/text">`

How are you planning to add href="" to cxl-icon-nav-item? I'm actually not sure how easy it would be to remove preventDefault() from vaadin-tab (so the inner link gets clicked)

Reply:
I configured staging menu CXL logo menu item to output `<vaadin-tab id="menu-item-1031802" class="menu__item menu__item--type-html">`

`menu__item--type-html` per https://themehybrid.com/weblog/why-mythic-uses-bem